templates/pg_dump.sh.epp

```diff
@@ -1,3 +1,20 @@
+<%- |
+  Boolean                                  $compress,
+  Array                                    $databases,
+  Optional[String[1]]                      $db_user,
+  Boolean                                  $delete_before_dump,
+  String[1]                                $dir,
+  Enum['plain','custom','directory','tar'] $format,
+  Array                                    $optional_args,
+  Optional[String[1]]                      $post_script,
+  Optional[String[1]]                      $pre_script,
```

Comment:
It calls `flatten($pre_script)` and `flatten($post_script)` which implies an `Array` is supported/expected. In addition to that, it may even be nested (given flatten is called). If you're doing a breaking change, I'd suggest `Array[String[1]]` and remove the `flatten()` call.

Reply:
see comment above